docs/docs/reference/custom-components/02-custom-component.md

```diff
@@ -554,6 +554,14 @@ Lifecycle components publish outputs only when the component is in the `ACTIVE`
 
 :::
 
+:::caution
+
+Each signal shall be named using only characters from the alphanumeric set (letters A-Z and numbers 0-9) and the
+underscore (`_`). Additionally, the `lower_snake_case` convention should be used and signal names may not start with a
+number or an underscore. 
+
+:::
```

Comment:
I would not bury this down at the end of the signals section. You also use capitals when you write `alphanumeric set (letters A-Z and numbers 0-9)`, but it would be preferable to document only `lower_snake_case` . I would rather structure and rewrite it like this:


```markdown
## Adding signals
[no change]

### Signal name

Component signals are declared and managed with unique signal names. The signal name is used to determine the default
topic. For a signal added under the name `foo`, the default topic will be `~/foo`. Adding a signal also automatically
creates and associates a parameter with that signal with the parameter name `[signal_name]_topic`, which is used to
override the default topic name at runtime. For a signal name `foo`, the parameter associated to the signal topic will
be called `foo_topic`.

For these reasons, signal names must adhere to the following rules:

- Signal names must be unique across all inputs and outputs
- Signal names must be written in `lower_snake_case` (using only lowercase letters, numbers and underscores)
- Signal names cannot start with a number or underscore.

## Message types
[no change]
```